### PR TITLE
Also replace paths in [build-dependencies]

### PR DESCRIPTION
**Pull request by en**
fix the following error:
```
$ ./scripts/node-template-release.sh substrate-node-template.tar.gz
    Finished dev [unoptimized + debuginfo] target(s) in 0.05s
     Running `target/debug/node-template-release /Users/ys/Desktop/polkadot/substrate/node-template /Users/ys/Desktop/polkadot/substrate/substrate-node-template.tar.gz`
error: failed to read `/private/var/folders/vm/yzhl_9_959gchmw245_0l4dw0000gn/T/.tmpIpC397/core/utils/build-script-utils/Cargo.toml`

Caused by:
  No such file or directory (os error 2)
thread 'main' panicked at 'assertion failed: Command::new("cargo").args(&["build",
                             "--all"]).current_dir(path).status().expect("Compiles node").success()', src/main.rs:165:2
```

### CODE REVIEW

File: scripts/node-template-release/src/main.rs [hunk 1]
```diff
@@ -88,33 +88,35 @@ fn replace_path_dependencies_with_git(cargo_toml_path: &Path, commit_id: &str, c
...
-			v.insert("git".into(), SUBSTRATE_GIT_URL.into());
-			v.insert("rev".into(), commit_id.into());
-
-			(k.clone(), v.into())
-		}).collect::<HashMap<_, _>>();
-
-	dependencies.extend(deps_rewritten.into_iter());
-
-	cargo_toml.insert("dependencies".into(), dependencies.into());
+	for table in vec!["dependencies", "build-dependencies"] {
```
Comment (bkchr) [inline scripts/node-template-release/src/main.rs]:
```suggestion
	for table in &["dependencies", "build-dependencies"] {
```

Comment (en) [reply scripts/node-template-release/src/main.rs]: I think an extra & is needed before _table_ if we do so.

Comment (bkchr) [reply scripts/node-template-release/src/main.rs]: No. Please revert this.

File: scripts/node-template-release/src/main.rs [hunk 2]
```diff
@@ -88,33 +88,35 @@ fn replace_path_dependencies_with_git(cargo_toml_path: &Path, commit_id: &str, c
...
-			v.insert("git".into(), SUBSTRATE_GIT_URL.into());
-			v.insert("rev".into(), commit_id.into());
-
-			(k.clone(), v.into())
-		}).collect::<HashMap<_, _>>();
-
-	dependencies.extend(deps_rewritten.into_iter());
-
-	cargo_toml.insert("dependencies".into(), dependencies.into());
+	for &table in &["dependencies", "build-dependencies"] {
```
Comment (bkchr) [inline scripts/node-template-release/src/main.rs]:
```suggestion
	for table in &["dependencies", "build-dependencies"] {
```

**Pull request merged**